src/test/java/io/nats/compatibility/ObjectStoreCommand.java

```diff
@@ -0,0 +1,280 @@
+// Copyright 2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.nats.compatibility;
+
+import io.nats.client.Connection;
+import io.nats.client.ObjectStore;
+import io.nats.client.ObjectStoreManagement;
+import io.nats.client.api.*;
+import io.nats.client.impl.NatsObjectStoreWatchSubscription;
+import io.nats.client.support.ApiConstants;
+import io.nats.client.support.JsonValueUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class ObjectStoreCommand extends Command {
+
+    final String bucket;
+    final String object;
+
+    public ObjectStoreCommand(Connection nc, TestMessage tm) {
+        super(nc, tm);
+        bucket = JsonValueUtils.readString(full, "bucket");
+        object = JsonValueUtils.readString(full, "object");
+    }
+
+    public void execute() {
+        switch (test) {
+            case DEFAULT_BUCKET:  doCreateBucket(); break;
+            case CUSTOM_BUCKET:   doCreateBucket(); break;
+            case GET_OBJECT:      doGetObject(); break;
+            case PUT_OBJECT:      doPutObject(); break;
+            case GET_LINK:        doGetLink(); break;
+            case PUT_LINK:        doPutLink(); break;
+            case UPDATE_METADATA: doUpdateMetadata(); break;
+            case WATCH:           doWatch(); break;
+            case WATCH_UPDATE:    doWatchUpdate(); break;
+            default:
+                respond();
+                break;
+        }
+    }
+
+    private void doCreateBucket() {
+        try {
+            _createBucket();
+            respond();
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doPutObject() {
+        try {
+            _createBucket();
+            String objectName = JsonValueUtils.readString(config, "name");
+            String description = JsonValueUtils.readString(config, "description");
+            _putObject(objectName, description);
+            respond();
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doGetObject() {
+        try {
+            _createBucket();
```

Comment:
You don't need to create state for the tests - compatibility tester does it for you. So you can remove `_createBucket()` and `_putObject()` here and in other tests and just do `_getObject()`. So this test can be reduced to:
```java
        try {
            ObjectStore os = nc.objectStore(bucket);
            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            _respondDigest(os.get(object, baos));
        }
```

Reply:
Same goes for all other tests preparing initial state: `doUpdateMetadata()`, `doPutObject()` etc.

Reply:
Fixing tests for state setup.